doc/basic_types.md

```diff
@@ -1,5 +1,360 @@
+```
+
+Used to create a Node Addon API `Value` that represents an **empty** value.
+
+```cpp
+Value(napi_env env, napi_value value); 
+```
+
+- `[in] env` - The `napi_env` environment in which to construct the Value object.
+- `[in] value` - The underlying JavaScript value that the `Value` instance represents
+
+Used to create a Node addon API `Value` that represents the `napi_value` passed
+in.
+
+### Operators
+
+#### operator napi_value
+
+```cpp
+operator napi_value() const;
+```
+
+Returns the underlying N-API value primitive. If the instance is _empty_,
+this returns  `nullptr`.
+
+#### operator ==
+
+```cpp
+bool operator ==(const Value& other) const;
+```
+
+Returns `true` if this value strictly equals another value, or `false` otherwise
+
+#### operator !=
+
+```cpp
+bool operator !=(const Value& other) const;
+```
+
+Returns `false` if this value strictly equals another value, or `true` otherwise
+
+### Methods
+
+#### From
+```cpp
+template <typename T>
+static Value From(napi_env env, const T& value);
+```
+
+- `[in] env` - The `napi_env` environment in which to construct the Value object.
+- `[in] value` - The C++ type to represent in JavaScript.
+
+This method is used to convert from a C++ type to a JavaScript value.
+Here, `value` may be any of:
+- `bool` - returns a `Napi::Boolean`
+- Any integer type - returns a `Napi::Number`
+- Any floating point type - returns a `Napi::Number`
+- `const char*` (encoded using UTF-8, null-terminated) - returns a `Napi::String`
+- `const char16_t*` (encoded using UTF-16-LE, null-terminated) - returns a `Napi::String`
+- `std::string` (encoded using UTF-8) - returns a `Napi::String`
+- `std::u16string` - returns a `Napi::String`
+- `napi::Value` - returns a `Napi::Value`
+- `napi_value` - returns a `Napi::Value`
+
+#### As
+```cpp
+template <typename T> T As() const;
+```
+
+Casts to another type of `Napi::Value`, when the actual type is known or assumed.
+
+This conversion does NOT coerce the type. Calling any methods inappropriate for 
+the actual value type will throw `Napi::Error`.
+
+#### StrictEquals
+```cpp
+bool StrictEquals(const Value& other) const;
+```
+
+- `[in] other` - The value to compare against
+
+Tests if this value strictly equals another value.
+
+#### Env
+```cpp
+Napi::Env Env() const;
+```
+
+Gets the environment the value is associated with. See `napi_env` for more 
+details about environments.
+
+#### IsEmpty
+```cpp
+bool IsEmpty() const;
+```
+
+Checks if the value is uninitialized.
+
+An empty value is invalid, and most attempts to perform an operation on an 
+empty value will result in an exception. Note an empty value is distinct from
+JavaScript `null` or `undefined`, which are valid values.
+
+When C++ exceptions are disabled at compile time, a method with a `Value` 
+return type may return an empty value to indicate a pending exception. So when 
+not using C++ exceptions, callers should check whether the value is empty 
+before attempting to use it.
```

Comment:
Maybe it's better to move this part on top and here: Returns `true` if the `Value` is an _empty_ value. 

About the error if the function that you are calling doesn't return value it' more appropriate use `Env::IsExceptionPending()` to retrieve if there is an active error.